e2e/LongHaul/LongHaul report/LongHaul Report.md

```diff
@@ -0,0 +1,164 @@
+
+![_](disconnectedDuration.png)
+
+### Telemetry Message Backlog
+
+Each device creates and sends telemetry messages to service every 3 seconds. If for some reason a message is not completed before its time to send a new telemetry message, the new message is stored in queue to be processed whenever possible. The graph below shows size of telemetry message backlog. The size of backlog queue always stayed at zero with few exceptions where it increased to size of 1 or 2. This is mostly due to when MQTT devices were reconnecting while a telemetry message was ready to be sent. Thus as soon as the device is connected again, backlogged telemetry messages are processed.
+
+![_](message_backlog.png)
+
+### Telemetry message delay
+
+Each device logs the time (seconds) it takes to successfully send a telemetry message to service. The graph below observes time < 1s to send telemetry message with exceptions where a device was reconnecting. None of such instances took more than 6 seconds.
+
+![_](telemetryDelay.png)
+
+### Reported twin update delay
+
+Each device logs the time (seconds) it takes to successfully receive reported twin updates. The graph below observes time < 1s to receive reported twin updates with exceptions where a device was reconnecting.
+
+![_](reportedTwinUpdateDelay.png)
+
+### Direct method delay
+
+Each device logs the time (seconds) it takes to successfully receive direct method from service. The graph below observes time < 1s to receive direct method with exceptions where a device was reconnecting and the delay was ~5-6 seconds.
+
+![_](DirectMethodDelay.png)
+
+### C2D message delay
+
+Each device logs the time (seconds) it takes to successfully receive c2d message from service. The graph below observes time < 1s to receive c2d message with exceptions where a device was reconnecting and the delay was ~5-6 seconds.
+
+![_](c2dDelay.png)
+
+
+## IoT Edge module client test results
+
+### Garbage collector available bytes
+
+The tests keep track of total available garbage collector bytes. This allows for detecting any memory leaks in the program. No significant memory leaks were observed during the tests. The screenshot below shows data from garbage collector over 2 edge modules used in the tests.
+The data shows consistent size of available garbage collector bytes.
+
+![_](moduleGCbytes.png)
+
+### Disconnection Duration
+
+MQTT protocol has a limitation where the device disconnects before token refresh and then reconnects. The device client sets TTL to 60 minutes by default, thus we see a reconnection after every ~55 minutes. The graph below shows the time duration (seconds) for which MQTT devices disconnects. Apart from a single exception, edge module reconnect within 1 second.
+
+![_](ModuleDisconnectionSeconds.png)
+
+### Telemetry Message Backlog
```

Comment:
Isn't this a repeat of what is on line 39?

Reply:
Oh, or a repeat for module client.

Reply:
line 39 is for iot hub device,. while this one is for edge module but yes, similar text for both.